### PR TITLE
Product switcher on mobile

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**
#### *My Products* label in popover menu is not aligned with its icon. This bug will be fixed in another PR.

The version of Product Switcher depends on which link was clicked to open it. This means it's dependent on the fact if shellbar is collapsed or not, not on a viewport size directly. With this approach, the Product Switcher won't be damaged by any further changes in shellbar's behaviour and the situation with having desktop menu and mobile Product Switcher (and opposite) is impossible.

Changes proposed in this pull request:

- add a link to mobile popover menu that opens mobile Product Switcher
- manage Product Switcher being desktop or mobile with Svelte state (not CSS)
- add a template for mobile Product Switcher
- add custom styles for full-width modal and list inside it [(see this)](https://github.com/SAP/fundamental/issues/1213)
- write mobile e2e tests

**Related issue(s)**

### CODE REVIEW

**Comment by y-kkamil** [comment]
please take a look at my comment above: 

> Shouldn't the mobile product switcher be closed after clicking on one of the elements? It seems more natural for me, rather than having to click on the item and then closing product switcher.

File: core/src/navigation/ProductSwitcher.html [hunk 1]
```diff
@@ -1,45 +1,95 @@
...
+                        {#each productSwitcherItems as productSwitcherItem}
+                          {#if productSwitcherItem.icon && productSwitcherItem.label}
+                          <li on:click="onActionClick(productSwitcherItem)" class="y-full-width-list__item">
+                              <span class="y-full-width-list__icon" style="background-image: url({productSwitcherItem.icon})" ></span>
+                              <span class="y-full-width-list__title">
+                                {productSwitcherItem.label}
+                              </span>
+                              <span class="y-full-width-list__subtitle">
+                                Subtitle 
+                                <!-- TODO: add proper subtitle -->
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/src/navigation/ProductSwitcher.html]: So.. should I add it to ProductSwitcher config?

File: core/src/navigation/TopNav.html [hunk 2]
```diff
@@ -11,135 +11,162 @@
...
-          </div> 
-      -->
+          </div>
+          {/if} 
+        {/each}
+        {#if isProductSwitcherAvailable}
+          <ProductSwitcher
+            bind:dropDownStates
+            on:toggleDropdownState="toggleDropdownState(event.name)"
+            on:setProductSwitcherToDesktop="set({isProductSwitcherMobile: false})"
```
Comment (dariadomagala) [inline core/src/navigation/TopNav.html]: This line is not needed anymore with the new approach.

File: core/src/navigation/TopNav.html [hunk 3]
```diff
@@ -11,135 +11,162 @@
...
+                          class="sap-icon--action sap-icon--s"
+                        ></span>
+                        {/if}
+                      </a>
+                    </li>
+                    {/if} 
+                    {/each}
+                    {#if isProductSwitcherAvailable}
+                      <li>       
+                        <a class="fd-menu__item" on:click="openMobileProductSwitcher(event)"  data-cy="mobile-product-switcher">
```
Comment (dariadomagala) [inline core/src/navigation/TopNav.html]: Now we could just leave for the click event `toggleDropdownState('productSwitcherPopover')`, nothing else is done in the `openMobileProductSwitcher` function anymore.

File: core/src/navigation/ProductSwitcher.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (dariadomagala) [reply core/src/navigation/ProductSwitcher.html]: I think so. Please check it with @hardl

File: core/src/navigation/TopNav.html (hunk 3 above)
Comment (parostatkiem) [reply core/src/navigation/TopNav.html]: You are wrong, the event propagation is stopped. That's why it doesn't work when I put `toggleDropdownState` here. But I did this change in `ProductSwitcher.html` where stopping propagation wasn't needed :)

File: core/src/navigation/ProductSwitcher.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/src/navigation/ProductSwitcher.html]: Anyway, I vote for a separate task here. If we introduced a new config variable, the docu would need to be updated and it'd be nice to use it in desktop PS also...

**Pull request merged**